cyctl/internal/create/template_auth_rules.go

```diff
@@ -0,0 +1,116 @@
+package create
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/cyclops-ui/cyclops/cyclops-ctrl/api/v1alpha1"
+	"github.com/cyclops-ui/cyclops/cyclops-ctrl/api/v1alpha1/client"
+	"github.com/cyclops-ui/cycops-cyctl/internal/kubeconfig"
+	"github.com/spf13/cobra"
+	v1Spec "k8s.io/api/core/v1"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+var (
+	createTemplateAuthRuleExample = ` 
+	# Create templateauth rule
+	cyctl create templateauthrule NAME --repo='github.com/repo/a' --username='name:key' --password='name:key'
+ 
+	# Create templateauthrule
+	cyctl create templateauthrule demo-templateauthrule --repo='https://github.com/cyclops-ui/templates' --username='name:john' --password='name:random'`
+)
+
+var (
+	username string
+	password string
+)
+
+// createTemplateAuthRule allows you to create create TemplateAuthRule Custom Resource.
+func createTemplateAuthRule(clientset *client.CyclopsV1Alpha1Client, templateAuthRuleName string) {
+	usernameSlice := strings.Split(username, ":")
+	passwordSlice := strings.Split(password, ":")
+
+	// Ensure key is present for both username and password
+	if len(usernameSlice) < 2 || usernameSlice[1] == "" || len(passwordSlice) < 2 || passwordSlice[1] == "" {
+		fmt.Printf("Invalid format for username or password. Expected 'name:key'.\n")
+		return
+	}
+
+	usernameKey, passwordKey := usernameSlice[1], passwordSlice[1]
+	var usernameName, passwordName string
+	if len(usernameSlice) > 1 {
+		usernameName = usernameSlice[0]
+	}
+	if len(passwordSlice) > 1 {
+		passwordName = passwordSlice[0]
+	}
+
+	if usernameName == "" || passwordName == "" {
+		fmt.Printf("Invalid format for username or password. Both name and key are required.\n")
+		return
+	}
```

Comment:
@siddhantprateek can you check it in the if statement above?

Reply:
i've added the changes, since the if statement was becoming quite lengthy, making it less readable, so i wrote it in separate function